src/algorithms/tracking/AmbiguitySolver.cc

```diff
@@ -0,0 +1,144 @@
+// SPDX-License-Identifier: LGPL-3.0-or-later
+// Copyright (C) 2024 Minjung Kim, Barak Schmookler
+#include "AmbiguitySolver.h"
+#include "AmbiguitySolverConfig.h"
+
+#include <Acts/AmbiguityResolution/GreedyAmbiguityResolution.hpp>
+#include <Acts/Definitions/TrackParametrization.hpp>
+#include <Acts/Definitions/Units.hpp>
+#include <Acts/EventData/MultiTrajectoryHelpers.hpp>
+#include <Acts/EventData/ParticleHypothesis.hpp>
+#include <Acts/EventData/SourceLink.hpp>
+#include <Acts/EventData/TrackContainer.hpp>
+#include <Acts/EventData/TrackStatePropMask.hpp>
+#include <Acts/EventData/TrackStateType.hpp>
+#include <Acts/EventData/VectorMultiTrajectory.hpp>
+#include <Acts/EventData/VectorTrackContainer.hpp>
+#include <ActsExamples/EventData/IndexSourceLink.hpp>
+#include <ActsExamples/EventData/Track.hpp>
+#include <ActsExamples/EventData/Trajectories.hpp>
+#include <boost/container/flat_set.hpp>
+#include <boost/container/vector.hpp>
+#include <edm4eic/Cov6f.h>
+#include <edm4eic/Measurement2DCollection.h>
+#include <edm4hep/Vector2f.h>
+#include <edm4hep/Vector3f.h>
+#include <fmt/core.h>
+#include <Eigen/Core>
+#include <algorithm>
+#include <array>
+#include <cmath>
+#include <cstddef>
+#include <optional>
+#include <utility>
+
+#include "Acts/Utilities/Logger.hpp"
+#include "extensions/spdlog/SpdlogFormatters.h" // IWYU pragma: keep
+#include "extensions/spdlog/SpdlogToActs.h"
+
+namespace eicrecon {
+
+using namespace Acts::UnitLiterals;
+
+// This array relates the Acts and EDM4eic covariance matrices, including
+// the unit conversion to get from Acts units into EDM4eic units.
+//
+// Note: std::map is not constexpr, so we use a constexpr std::array
+// std::array initialization need double braces since arrays are aggregates
+// ref: https://en.cppreference.com/w/cpp/language/aggregate_initialization
+static constexpr std::array<std::pair<Acts::BoundIndices, double>, 6> edm4eic_indexed_units{
+    {{Acts::eBoundLoc0, Acts::UnitConstants::mm},
+     {Acts::eBoundLoc1, Acts::UnitConstants::mm},
+     {Acts::eBoundPhi, 1.},
+     {Acts::eBoundTheta, 1.},
+     {Acts::eBoundQOverP, 1. / Acts::UnitConstants::GeV},
+     {Acts::eBoundTime, Acts::UnitConstants::ns}}};
+
+Acts::GreedyAmbiguityResolution::Config
+transformConfig(const eicrecon::AmbiguitySolverConfig& cfg) {
+  Acts::GreedyAmbiguityResolution::Config result;
+  result.maximumSharedHits = cfg.maximum_shared_hits;
+  result.maximumIterations = cfg.maximum_iterations;
+  result.nMeasurementsMin  = cfg.n_measurements_min;
+  return result;
+}
+
+std::size_t sourceLinkHash(const Acts::SourceLink& a) {
+  return static_cast<std::size_t>(a.get<ActsExamples::IndexSourceLink>().index());
+}
+
+bool sourceLinkEquality(const Acts::SourceLink& a, const Acts::SourceLink& b) {
```

Comment:
```suggestion
static bool sourceLinkEquality(const Acts::SourceLink& a, const Acts::SourceLink& b) {
```
Avoid possible clashes in symbol space